triplestore-jena-memory/src/main/java/zone/cogni/asquare/triplestore/pool/LocalTdbRdfStoreServicePool.java

```diff
@@ -0,0 +1,146 @@
+package zone.cogni.asquare.triplestore.pool;
+
+import org.apache.commons.pool2.KeyedObjectPool;
+import org.apache.commons.pool2.PoolUtils;
+import org.apache.commons.pool2.impl.AbandonedConfig;
+import org.apache.commons.pool2.impl.GenericKeyedObjectPool;
+import org.apache.commons.pool2.impl.GenericKeyedObjectPoolConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zone.cogni.asquare.triplestore.jenamemory.LocalTdbRdfStoreService;
+import zone.cogni.asquare.triplestore.pool.factory.LocalTdbRdfStoreServiceFactory;
+import zone.cogni.asquare.triplestore.pool.key.LocalTdbPoolKey;
+
+import java.io.Closeable;
+import java.time.Duration;
+
+public final class LocalTdbRdfStoreServicePool implements Closeable {
+
+  private static final Logger log = LoggerFactory.getLogger(LocalTdbRdfStoreServicePool.class);
+
+  private final GenericKeyedObjectPool<LocalTdbPoolKey, LocalTdbRdfStoreService> genericPool;
+  private final KeyedObjectPool<LocalTdbPoolKey, LocalTdbRdfStoreService> pool;
```

Comment:
I do not get the difference between the 2 pools

Reply:
genericPool is the not decorated pool and it is needed to be stored because it has some useful methods that we don't have on the interface, the `pool` is the final result, the decorated pool